### PR TITLE
Add project deletion task for composite views

### PR DESCRIPTION
**Pull request by imsdu**
Related to #3049

### CODE REVIEW

File: delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  private val viewStream: Stream[Task, CompositeViewDef] =
+    Stream(
+      activeView(active1),
+      DeprecatedViewDef(deprecated),
+      activeView(active2)
+    )
+
+  test("Deprecate all active views for project") {
+    for {
+      deprecated   <- Ref.of[Task, Set[ViewRef]](Set.empty)
```
Comment (shinyhappydan) [inline delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala:43]: Small thing, but perhaps this should be a list? Not sure if duplication would indicate an issue

File: delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTask.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+import ch.epfl.bluebrain.nexus.delta.plugins.compositeviews.indexing.CompositeViewDef.{ActiveViewDef, DeprecatedViewDef}
+import ch.epfl.bluebrain.nexus.delta.sdk.deletion.ProjectDeletionTask
+import ch.epfl.bluebrain.nexus.delta.sdk.deletion.model.ProjectDeletionReport
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sourcing.model.ProjectRef
+import com.typesafe.scalalogging.Logger
+import fs2.Stream
+import monix.bio.{Task, UIO}
+
+final class CompositeViewsDeletionTask(
```
Comment (shinyhappydan) [inline delta/plugins/composite-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTask.scala:15]: I think it's fine without, but you would usually have a doc here

**Review by olivergrabinski — DISMISSED** [review]

File: delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala (hunk 1 above)
Comment (imsdu) [reply delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala:43]: Duplication can't happen here as view ref is the identifier of the view and we only process each view only once.

Comment (shinyhappydan) [reply delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala:43]: Well, the set here represents calls to the function, and you could call the function twice if you had a bug. I agree it's unlikely, but List would be more accurate than Set

**Review by shinyhappydan — APPROVED** [review]

File: delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala (hunk 1 above)
Comment (imsdu) [reply delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala:43]: Things can only be deprecated once

Comment (shinyhappydan) [reply delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala:43]: `deprecate(view, subject) >> deprecate(view, subject)` would be a bug but this test wouldn't fail

Comment (imsdu) [reply delta/plugins/composite-views/src/test/scala/ch/epfl/bluebrain/nexus/delta/plugins/compositeviews/deletion/CompositeViewsDeletionTaskSuite.scala:43]: The deprecate function in the test is idempotent just like the real one (and the other deletion tasks), so it wouldn't be a bug.

**Pull request merged**